### ISSUE TITLE
Typoscript language overlay trough _LOCAL_LANG doesn't work in pi2

### ISSUE THREAD

**Issue by yauhensk** [teaminmedias-pluswerk/ke_search#137]
This is because of `$typoScriptService->convertTypoScriptArrayToPlainArray` call before `$this->pi_loadLL('EXT:ke_search/pi1/locallang.xml');` so, `$this->conf['_LOCAL_LANG.']` is NULL, but `$this->conf['_LOCAL_LANG']` contains language labels.

**Comment by dr-bock** [comment]
Can confirm this bug, too.

**Comment by demonking** [comment]
Can also confirm this...
It's very annoying, because I need to set up a Site with > 4 languages ...

**Comment by plojewski** [comment]
When could you fix it please?

**Comment by mschwemer** [comment]
Same here ... as a workaround, you can use 

`$GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:ke_search/pi1/locallang.xml'][] = 'EXT:your-provider-ext/Resources/Private/Language/locallang_kesearch-pi1.xml';`

and provide the translated labels in an own xml file.

**Comment by eyemiru** [comment]
Still won't work, same issue even with the workaround

**Comment by Neowar** [comment]
We can confirm this issue also in pi1 in current version on TYPO3 8.7. 

With a litte correction in the code (see patch codes below) it works. This load the original $conf, run core functions for PiVars and Locallang and after that convert the conf array for internal ke_search functions.

Please adapt this patch codes in the current master.

Patch for pi1
```php
Index: typo3conf/ext/ke_search/pi1/class.tx_kesearch_pi1.php
===================================================================
--- typo3conf/ext/ke_search/pi1/class.tx_kesearch_pi1.php	(revision f2536682cea11bfbd3e6aa36d09fd2b5e692b489)
+++ typo3conf/ext/ke_search/pi1/class.tx_kesearch_pi1.php	(date 1541506778975)
@@ -54,9 +54,10 @@
             $typoScriptService = GeneralUtility::makeInstance(TYPO3\CMS\Extbase\Service\TypoScriptService::class);
         }
 
-        $this->conf = $typoScriptService->convertTypoScriptArrayToPlainArray($conf);
+        $this->conf = $conf;
         $this->pi_setPiVarDefaults();
         $this->pi_loadLL();
+        $this->conf = $typoScriptService->convertTypoScriptArrayToPlainArray($conf);
 
         // Configuring so caching is not expected. This value means that no cHash params are ever set.
         // We do this, because it's a USER_INT object!

```

Patch for pi2
```php
Index: typo3conf/ext/ke_search/pi2/class.tx_kesearch_pi2.php
===================================================================
--- typo3conf/ext/ke_search/pi2/class.tx_kesearch_pi2.php	(revision 9fef57cdd540ac50bc61308b54472fd6267de7c2)
+++ typo3conf/ext/ke_search/pi2/class.tx_kesearch_pi2.php	(date 1541508558607)
@@ -55,13 +55,15 @@
             $typoScriptService = GeneralUtility::makeInstance(TYPO3\CMS\Extbase\Service\TypoScriptService::class);
         }
 
-        $this->conf = $typoScriptService->convertTypoScriptArrayToPlainArray($conf);
+        $this->conf = $conf;
         $this->pi_setPiVarDefaults();
 
         // use pi1 locallang values, since all the frontend locallang values for
         // pi1, pi2 and pi3 are set in pi1 language file
         $this->pi_loadLL('EXT:ke_search/pi1/locallang.xml');
 
+        $this->conf = $typoScriptService->convertTypoScriptArrayToPlainArray($conf);
+
         // Configuring so caching is not expected. This value means that no cHash params are ever set.
         // We do this, because it's a USER_INT object!
         $this->pi_USER_INT_obj = 1;

```

**Comment by nikkialgar** [comment]
Thank you, this works for me.

**Comment by DavidBruchmann** [comment]
The hint of @mschwemer does work also in newer versions, just with adjusted path and filenames.
```
$GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:ke_search/Resources/Private/Language/locallang_searchbox.xml'][] = 'EXT:' . $_EXTKEY . '/Resources/Private/Language/ext/ke_search/locallang_searchbox.xml';
$GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:ke_search/Resources/Private/Language/locallang_resultlist.xml'][] = 'EXT:' . $_EXTKEY . '/Resources/Private/Language/ext/ke_search/locallang_resultlist.xml';
```

**Comment by DavidBruchmann** [comment]
If the code above is placed in the file `AdditionalConfiguration.php`, then the extension variable `$_EXTKEY` is not available and the extension has to be noted explicitly:
```
$GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:ke_search/Resources/Private/Language/locallang_searchbox.xml'][] = 'EXT:sitepackage/Resources/Private/Language/ext/ke_search/locallang_searchbox.xml';
$GLOBALS['TYPO3_CONF_VARS']['SYS']['locallangXMLOverride']['EXT:ke_search/Resources/Private/Language/locallang_resultlist.xml'][] = 'EXT:sitepackage/Resources/Private/Language/ext/ke_search/locallang_resultlist.xml';
```

**Comment by christianbltr** [comment]
This has been fixed in version 3.7.0.
